nexus/reconfigurator/planning/src/planner.rs

```diff
@@ -1595,6 +1596,49 @@ impl<'a> Planner<'a> {
                 }
                 true
             }
+            ZoneKind::BoundaryNtp => {
+                debug!(
+                    self.log,
+                    "Checking if boundary NTP zone can be shut down"
+                );
+
+                // Find all boundary NTP zones expected by our parent blueprint.
+                let mut boundary_ntp_zones = std::collections::HashSet::new();
+                for sled_config in
+                    self.blueprint.parent_blueprint().sleds.values()
```

Comment:
I think we want the currently-being-planned blueprint, not the parent, right? Expungement planning is the first thing we do, so we could have _just_ expunged a boundary NTP zone (or even a sled), and we wouldn't want to count those here.

Reply:
agh, yup, updated to use the "blueprint builder" methods to act on the under-construction blueprint